modules/marshaller-common/src/main/java/org/apache/ignite/internal/marshaller/Marshaller.java

```diff
@@ -267,13 +306,36 @@ public Object readObject(MarshallerReader reader, Object target) throws Marshall
             return obj;
         }
 
+        /** {@inheritDoc} */
+        @Override
+        public Object readValue(MarshallerReader reader, Object target) throws MarshallerException {
```

Comment:
not sure I fully understand the difference between `readObject` and `readValue`...

Reply:
Fixed.